src/grids.jl

```diff
@@ -251,6 +313,43 @@ get_colat(::Type{<:FullGaussianGrid},nlat_half::Integer) =
 get_quadrature_weights(::Type{<:FullGaussianGrid},nlat_half::Integer) = gaussian_weights(nlat_half)
 full_grid(::Type{<:FullGaussianGrid}) = FullGaussianGrid    # the full grid with same latitudes
 
+"""
+    G = FullHEALPix4Grid{T}
+
+A full HEALPix4 grid is a regular latitude-longitude grid that uses `nlat` HEALPix4 latitudes,
+and the same `nlon` longitudes for every latitude ring. The grid points are closer in zonal direction
+around the poles. The values of all grid points are stored in a vector field `v` that unravels
+the data 0 to 360˚, then ring by ring, which are sorted north to south."""
+struct FullHEALPix4Grid{T} <: AbstractFullGrid{T}
+    data::Vector{T}    # data vector, ring by ring, north to south
+    nlat_half::Int  # number of latitudes on one hemisphere
+
+    FullHEALPix4Grid{T}(data,nlat_half) where T = length(data) == npoints_fullhealpix4(nlat_half) ?
+    new(data,nlat_half) : error("$(length(data))-element Vector{$(eltype(data))} cannot be used to create a "*
+        "F$nlat_half ($(4nlat_half)x$(2nlat_half - 1)) FullHEALPix4Grid{$T}.")
```

Comment:
Agreed. As we expand the number of grids, there is clearly not obvious way to match each grid with one letter.
When I used J for HEALPix4, I had J for [juggling bean bag](https://commons.wikimedia.org/wiki/File:Juggling_balls.jpg) in my mind.
I'm tempted to group HEALPix4Grid with OctahedralGrid rather than with HEALPix12, though.